### PR TITLE
[WIP] wxpython: Fix build error

### PR DESCRIPTION
**Pull request by stahta01**
Something in the build environment changed and it broke package build.

Edit rewrote waf configure and build to fix build error.

Tim S.

### CODE REVIEW

**Comment by MehdiChinoune** [comment]
We have fixed shebang so it no longer requires MSYS python to be installed.
You should fix building wxPython with MINGW python instead of using the MSYS one.

**Comment by stahta01** [comment]
I have made it slightly farther into the waf configure step; but, still a lot to do.

Tim S.

File: mingw-w64-wxPython/PKGBUILD [hunk 1]
```diff
@@ -18,22 +18,25 @@ depends=("${MINGW_PACKAGE_PREFIX}-python"
          "${MINGW_PACKAGE_PREFIX}-python-pillow"
          "${MINGW_PACKAGE_PREFIX}-python-six"
          "${MINGW_PACKAGE_PREFIX}-wxmsw${_wx_basever}")
-makedepends=("${MINGW_PACKAGE_PREFIX}-doxygen"
+makedepends=('patch'
```
Comment (MehdiChinoune) [inline mingw-w64-wxPython/PKGBUILD:21]: `patch` is part of base-devel, no need to list under `makedepends`

File: mingw-w64-wxPython/PKGBUILD [hunk 2]
```diff
@@ -71,17 +78,26 @@ build() {
       --no_allmo --no_magic --regenerate_sysconfig \
         dox touch etg sip
 
+  echo "MSYSTEM := $MSYSTEM"
+
   local _jobs=${MAKEFLAGS:--j1}
   MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
-  CC_NAME=${CC} CXX_NAME=${CXX} \
+  CC=${MINGW_PREFIX}/bin/cc.exe \
+  CXX=${MINGW_PREFIX}/bin/${CXX}.exe \
```
Comment (MehdiChinoune) [inline mingw-w64-wxPython/PKGBUILD:86]:
```suggestion
  CXX=${MINGW_PREFIX}/bin/c++.exe \
```

File: mingw-w64-wxPython/PKGBUILD (hunk 1 above)
Comment (stahta01) [reply mingw-w64-wxPython/PKGBUILD:21]: Since, you moved the directions so many times I have no idea what packages should normally be installed!

Tim S.

Comment (MehdiChinoune) [reply mingw-w64-wxPython/PKGBUILD:21]: https://www.msys2.org/dev/new-package/#building-the-package

**Pull request closed without merging**